### PR TITLE
kate: rebuild for new version libgit2

### PR DESCRIPTION
**Pull request by WhiredPlanck**
Topic Description
-----------------

As `kate` and `kwrite` is actually depend on `libgit2` directly, which has been updated by #3460 , now they need to be rebuilt.

Package(s) Affected
-------------------

`kate` and `kwrite`

Security Update?
----------------

Test Build(s) Done
------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`
    

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

### CODE REVIEW

**Review by KexyBiscuit — CHANGES_REQUESTED** [review]
Please add `libgit2` to dependencies list.

**Review by MingcongBai — CHANGES_REQUESTED** [review]
You should also mark `kate` and `kwrite` in `libgit2`'s PKGBREAK.

**Comment by WhiredPlanck** [comment]
@KexyBiscuit @MingcongBai Roger. Just need to take a moment.

File: extra-libs/libgit2/autobuild/defines [hunk 1]
```diff
@@ -4,7 +4,8 @@ PKGDEP="zlib openssl libssh2 http-parser pcre2"
 PKGDES="A linkable library for Git"
 PKGBREAK="cargo-audit<=0.15.0-1 calligra<=3.2.0 ciel<=3.0.12 exa<=0.10.1-3 \
           fritzing<=0.9.6 geany-plugins<=1.37 gnome-builder<=3.40.2-1 \
-          ktexteditor<=5.86.0 libgit2-glib<=0.28.0.1-1 pygit2<=1.0.3-1"
+          ktexteditor<=5.86.0 kate<=21.08.1 kwirte<=21.08.1 \
+          libgit2-glib<=0.28.0.1-1 pygit2<=1.0.3-1"
```
Comment (MingcongBai) [inline extra-libs/libgit2/autobuild/defines:8]: You'd need to bump REL for libgit2.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-libs/libgit2/autobuild/defines (hunk 1 above)
Comment (WhiredPlanck) [reply extra-libs/libgit2/autobuild/defines:8]: Ok ~

**Comment by WhiredPlanck** [comment]
@MingcongBai I built `kwrite` after building `libgit2` v1.3.0, but `kwrtie` still links to `libgit2.so.28` instead of  `libgit2.so.1.3.0`. Totally have no idea ......

**Comment by MingcongBai** [comment]
@WhiredPlanck Could you make sure you updated your Git tree such that libgit2 has a bumper REL? If the version of libgit2 built matches the one on the repository, Ciel will not update it before the next package build.

**Comment by WhiredPlanck** [comment]
@JeffBai Sorry, I made a big mistake.  `kwrite` and `kate` just depend on `ktexteditor`, which actually depends on `libgit2`. I put `ktexteditor` in front of `libgit2` to build accidentally. Anyway, this is not necessary any more... I'll close it with this comment.

**Pull request closed without merging**